doc/contributing.md

```diff
@@ -16,37 +16,217 @@ and we encourage all to read it carefully.
+running linters, etc. If you don't want to use `make`, open the `Makefile` and
+copy the commands you want to run.
+
+To install the current source code into your testing environment, run:
+
+```bash
+make install
+```
 
-## Reporting a Bug
+This installs your project in *editable* mode, meaning that changes made to the source
+code will be available when you import the package (even if you're on a different
+directory).
 
-Find the *Issues* tab on the top of the GitHub repository and click *New Issue*.
-You'll be prompted to choose between different types of issue, like bug reports and
-feature requests.
-Choose the one that best matches your need.
-The Issue will be populated with one of our templates.
-**Please try to fillout the template with as much detail as you can**.
-Remember: the more information we have, the easier it will be for us to solve your
-problem.
+### Pull Request Workflow
 
+We follow the [git pull request workflow](http://www.asmeurer.com/git-workflow)
+to make changes to our codebase. Every change made goes through a pull request, even
+our own, so that our
+[continuous integration](https://en.wikipedia.org/wiki/Continuous_integration)
+services have a chance to check that the code is up to standards and passes all
+our tests. This way, the *master* branch is always stable.
 
-## Editing the Documentation
+#### General Guidelines for Making a Pull Request (PR):
+
+* What should be included in a PR
+  - Have a quick look at the titles of all the existing issues first. If there
+    is already an issue that matches your PR, leave a comment there to let us
+    know what you plan to do. Otherwise, **open an issue** describing what you
+    want to do.
+  - Each pull request should consist of a **small** and logical collection of
+    changes; larger changes should be broken down into smaller parts and
+    integrated separately.
+  - Bug fixes should be submitted in separate PRs.
+* How to write and submit a PR
+  - Use underscores for all Python (*.py) files as per
+    [PEP8](https://www.python.org/dev/peps/pep-0008/), not hyphens. Directory
+    names should also use underscores instead of hyphens.
+  - Describe what your PR changes and *why* this is a good thing. Be as
+    specific as you can. The PR description is how we keep track of the changes
+    made to the project over time.
+  - Do not commit changes to files that are irrelevant to your feature or
+    bugfix (e.g.: `.gitignore`, IDE project files, etc).
+  - Write descriptive commit messages. Chris Beams has written a
+    [guide](https://chris.beams.io/posts/git-commit/) on how to write good
+    commit messages.
+* PR review
+  - Be willing to accept criticism and work on improving your code; we don't
+    want to break other users' code, so care must be taken not to introduce
+    bugs.
+  - Be aware that the pull request review process is not immediate, and is
+    generally proportional to the size of the pull request.
+
+#### Pull Request Review Process
```

Comment:
To be consistent with **General Guidelines for Making a Pull Request (PR):*** above? :smile:
```suggestion
#### General Process for Pull Request Review:
```